File: leet_easy_c++/leet_001/leet_001b.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        vector<int> sol(2,0);
        int tmp, len = nums.size();
        for(int i=0; i<len; i++) umap[*(nums.begin()+i)] = i;
        for(int i=0; i<len; i++){
            tmp = target - *(nums.begin()+i);
            if(tmp<<1 != target){
                if(umap.find(tmp) != umap.end()){
                    sol[0] = i;
                    sol[1] = umap[tmp];
                    return sol;
                }   
            }
            else if(umap.find(tmp) != umap.end()){
                if (umap[tmp] != i){
                    sol[0] = i;
                    sol[1] = umap[tmp];
                    return sol;
                }
            }
        }
        return sol; //should never get here given specs 
    }
};

File: leet_easy_java/leet_001/leet_001b.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len = nums.length;
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        for(int i=0; i< len; i++){
            if(hmap.get(target - nums[i]) != null){
                return new int[]{hmap.get(target - nums[i]), i};
            } 
            hmap.put(nums[i], i);
        }
        return new int[2];
    }
}